app/models/host/base.rb

```diff
@@ -609,8 +609,11 @@ def uniq_interfaces_identifiers
     end
 
     def password_base64_encrypted?
-      if root_pass_changed?
-        root_pass == hostgroup.try(:read_attribute, :root_pass)
+      hostgroup_root_pass = hostgroup.try(:read_attribute, :root_pass)
+      if self[:root_pass].nil? && hostgroup_root_pass.nil?
```

Comment:
Should I rather check `.absent?` instead of `.nil?`?

Reply:
I couldn't find `absent?`, but I would go for `blank?`, as it covers most of the cases:
```ruby
nil.blank? => true
''.blank? => true
[].blank? => true
{}.blank? => true
```

Reply:
Yeah, for some reason I have that method named wrong in my mind :facepalm: